blivetgui/blivet_utils.py

```diff
@@ -633,22 +637,68 @@ def device_resizable(self, blivet_device):
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
                                       max_size=blivet_device.size)
 
         elif not blivet_device.format.type:
+            # unformatted devices are not resizable (except extended partitions)
             if (blivet_device.type == "partition" and blivet_device.is_extended and (blivet_device.max_size > blivet_device.size or
                blivet_device.min_size < blivet_device.size)):
                 return ProxyDataContainer(resizable=True, error=None, min_size=blivet_device.min_size,
                                           max_size=blivet_device.max_size)
             else:
-                return ProxyDataContainer(resizable=False, error=None, min_size=blivet.size.Size("1 MiB"),
+                msg = _("Unformatted devices are not resizable")
+                return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
                                           max_size=blivet_device.size)
 
-        if blivet_device.type in ("lvmlv",) and self._has_snapshots(blivet_device):
-            msg = _("Logical Volumes with snapshots couldn't be resized.")
+        elif blivet_device.format.type not in ("ext2", "ext3", "ext4", "ntfs"):
+            # unfortunately we can't use format._resizable here because blivet uses it to both mark
+            # formats as not resizable and force users to call update_size_info on resizable formats
+            msg = _("Resizing of {type} format is currently not supported").format(type=blivet_device.format.type)
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
+                                      max_size=blivet_device.size)
+
+        elif not blivet_device.format._resize.available:
+            msg = _("Tools for resizing format {type} are not available.").format(type=blivet_device.format.type)
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
+                                      max_size=blivet_device.size)
+
+        elif blivet_device.protected:
+            msg = _("Protected devices cannot be resized")
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
+                                      max_size=blivet_device.size)
+
+        elif blivet_device.format_immutable:
+            msg = _("Immutable formats cannot be resized")
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
+                                      max_size=blivet_device.size)
+
+        elif blivet_device.children:
+            msg = _("Devices with children cannot be resized")
+            return ProxyDataContainer(resizable=False, error=msg, min_size=blivet.size.Size("1 MiB"),
+                                      max_size=blivet_device.size)
```

Comment:
I'd check `protected` and `children` before checking `format.type` and `format._resize.available` because those reasons are "stronger".

Reply:
Makes sense, fixed.